docs/src/scraper.py

```diff
@@ -0,0 +1,95 @@
+import os
+import shutil
+
+from sphinx_gallery.scrapers import matplotlib_scraper
+
+class SphinxGalleryScraper:
+    def __repr__(self):
+        return 'SphinxGalleryScraper'
+
+    def __call__(self, block, block_vars, gallery_conf):
+        variables = block_vars.get("example_globals", {})
+        widget = variables.get("___")
+        if widget:
+            # Copy js scripts
+            target_dir = gallery_conf.get("gallery_dirs")
+            if not target_dir:
+                raise ValueError("gallery_conf does not contain 'gallery_dirs' key.")
+            source_files = [
+                "./_static/js/chemiscope.min.js",
+                "./_static/js/chemischope-sphinx-gallery.js"
+            ]
+            self.copy_chemiscope_files(target_dir, source_files)
+
+            # Use the custom html content
+            widget._repr_html_ = lambda: self.generate_html_content(widget.value)
+
+        # Use sphinx-gallery standard scraper
+        return matplotlib_scraper(block, block_vars, gallery_conf)
```

Comment:
do we need to do this? since the matplotlib scrapper is already in the list of scrappers in the conf